jasmine/spec/inverted-index-test.js

```diff
@@ -1,10 +1,84 @@
-// asserting book.json is not empty
-var obj = new Index();
-obj.createIndex('../../books.json');
-var contents = obj.fileContents;
-
-describe("Read book data", function () {
-  it("Checks to assert that book.json is not empty", function () {
-    expect(contents.length).not.toBe(0);
+describe("Inverted Index Tests", function () {
+  var index = new Index();
+
+  // perform this before the start of each test
+  beforeEach(function (done) {
+    index.readFile('/books.json').then(function (data) {
+      index.results = data;
+      index.getIndex();
+      done();
+    });
+  });
+
+  describe("Read book data", function () {
+    it("Checks to assert that book.json is not empty", function () {
+      expect(typeof index.results).toBe('string');
+      expect(index.results.length).not.toEqual(0);
+    });
+
+    it("should check if input file is a valid json object", function () {
+      expect(index.fileContents).not.toBeUndefined();
+    });
+
+    it("should check if the input file contains an array of objects", function () {
+      expect(Array.isArray(index.fileContents)).toBe(true);
+      expect(index.fileContents[0] instanceof Object).toBe(true);
+      expect(index.fileContents[1] instanceof Object).toBe(true);
+    });
+  });
+
+  describe("Populate Index", function () {
+    it("checks if getIndex method returns an object", function () {
+      expect(typeof index.getIndex()).toBe('object');
+    });
+
+    it("checks if the index created is accurate", function () {
+      expect(index.getIndex().alice).toEqual([0]);
+      expect(index.getIndex().ring).toEqual([1]);
+      expect(index.getIndex().njerrywerry).toEqual(undefined);
+    });
+  });
+
+  describe("Search Index", function () {
+    it("checks if searchIndex method returns an array", function () {
+      expect(Array.isArray(index.searchIndex('alice'))).toBe(true);
+    });
+
+    it("checks if the results of the search are correct", function () {
+      expect(index.searchIndex('alice')).toContain(0);
```

Comment:
**Code Quality - Critical**

You are checking whether the result contains 0 meaning if the result is `['hey I have 0 chills', 0]`, this spec will pass though the result is not correct.